src/main/java/org/elasticsearch/action/admin/cluster/repositories/verify/TransportVerifyRepositoryAction.java

```diff
@@ -69,7 +69,7 @@ protected VerifyRepositoryResponse newResponse() {
 
     @Override
     protected ClusterBlockException checkBlock(VerifyRepositoryRequest request, ClusterState state) {
-        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA, "");
+        return state.blocks().indexBlockedException(ClusterBlockLevel.METADATA_WRITE, "");
```

Comment:
During repository verification proces, each node will try to write a test file into the target repository and then deleting it. That's why METADATA_WRITE seems more appropriate to me.